### ISSUE TITLE
The speed performance of validation_batch_size does not work.

### ISSUE THREAD

**Issue by yananchen1989** [tensorflow/tensorflow#48953]
I assume that the `validation_batch_size` could help speed up the validation process at each epoch. However, this contradict with the experiments.

> In [15]: x_train.shape
> Out[15]: (128, 1)
> 
> In [16]: x_test.shape
> Out[16]: (7600, 1)

Here is my experiments that I run several times, tf version: 2.3.1
```
model = get_model_transormer(num_classes)
for b in [8, 32, 64, 256, 512, 1024]:
    t1 = time.time()
    history = model.fit(
        x_train, y_train, batch_size=8, epochs=1, \
        validation_batch_size=b,
        validation_data=(x_test, y_test), verbose=1,
        callbacks = [EarlyStopping(monitor='val_acc', patience=3, mode='max')]
    )
    t2 = time.time()
    print('b:',b,  ' diff:', t2-t1)
```
Here is the report:

> 16/16 [==============================] - 12s 734ms/step - loss: 0.6992 - acc: 0.7812 - val_loss: 1.1789 - val_acc: 0.4779
> b: 8  diff: 11.802898168563843
> 16/16 [==============================] - 7s 468ms/step - loss: 0.5877 - acc: 0.7891 - val_loss: 1.4876 - val_acc: 0.3555
> b: 32  diff: 7.545728445053101
> 16/16 [==============================] - 7s 420ms/step - loss: 0.4992 - acc: 0.8594 - val_loss: 1.4072 - val_acc: 0.3999
> b: 64  diff: 6.773781061172485
> 16/16 [==============================] - 12s 731ms/step - loss: 0.3447 - acc: 0.9062 - val_loss: 1.1698 - val_acc: 0.5408
> b: 256  diff: 11.749623537063599
> 16/16 [==============================] - 27s 2s/step - loss: 0.2456 - acc: 0.9453 - val_loss: 1.0378 - val_acc: 0.5968
> b: 512  diff: 26.67411971092224
> 16/16 [==============================] - 52s 3s/step - loss: 0.1412 - acc: 0.9922 - val_loss: 1.1378 - val_acc: 0.5218
> b: 1024  diff: 52.29804968833923

It seems that setting validation_batch_size to 64 is the optimal option. That's quite weird.
Any explanation ?

**Comment by UsharaniPagadala** [comment]
@yananchen1989 

We see that the issue template has not been filled, could you please do so as it helps us analyse the issue [tf version, steps followed before you ran into this error or stand alone code to reproduce the issue faced]
Thanks
